CMakeLists.txt

```diff
@@ -57,16 +57,33 @@ set(CMAKE_CXX_VISIBILITY_PRESET hidden)
 set(CMAKE_VISIBILITY_INLINES_HIDDEN TRUE)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
 
+# Conan dependencies
+include(conan)
+set(DYNAMIC_LIB_TRUE_FALSE "False")
+if(${DYNAMIC_LIB})
+  set(DYNAMIC_LIB_TRUE_FALSE "True")
+endif()
+
+# Detect compiler, build type etc from cmake settings
+conan_cmake_autodetect(conan_settings)
```

Comment:
Conan gets some details of your build environment such as compiler, and built type from cmake settings. There are alternative ways of doing this, for example using profile files: https://docs.conan.io/en/latest/reference/profiles.html

Reply:
Unless there are settings you can control with those files but not through CMake directly, I would say we should stick with the solution you picked. Otherwise we have to duplicate settings.

Reply:
The approach here would respect cmake flags such as `CMAKE_CXX_COMPILER`?